### DISCUSSION TITLE
How to write a freaking cool README

### DISCUSSION THREAD

**Discussion by mikaeloduh** [Game-as-a-Service/Lobby-Platform-Service#97] [General]
**A cool project readme must have:**

- Well-documented
- Internationalize
- Live demo
- A bunch of badges
- Sophisticated technology stack
- Looks very active
- Content-rich dashboard 
- Everything is on Git
  - Infrastructure as code
  - Discussion
  - Agenda
- Amazing contributors (that's us!)

**Comment by ricksu978** [comment]
# 目標
讓大家的履歷可以有一個 freaking awesome 的開源專案，並在面試外商時上的了檯面。

# 我們的 readme 需要怎麼寫呢?
大家可以多多觀察世界級的開源專案，記錄下他們的 readme 有甚麼特點?
並在下面留言，我們會整理再來展開細節。

- [ ] Code of Conduct
- [ ] Contributing Guide
- [ ] Where to Get Support
- [ ] Multilingual (English and Chinese)
- [ ] Links to live environment
- [ ] Badges (CI / CD status ... etc)
- [ ] Technology Stack

**Reply by arealclimber** [reply]
有明確的 Code of Conduct 跟 Contributing Guide 跟 Where to Get Support，參考 [react](https://github.com/facebook/react) 跟 [playwright](https://github.com/microsoft/playwright) 跟 [stable-diffusion-webui](https://github.com/AUTOMATIC1111/stable-diffusion-webui)

**Comment by ricksu978** [comment]
> * Content-rich dashboard

@mikaeloduh 
Could you give an example of a `Content-rich dashboard` ?

**Reply by mikaeloduh** [reply]
Like this https://github.com/DevOps-Nirvana/Grafana-Dashboards

**Comment by mikaeloduh** [comment]
### 今日會議討論 (6/25)：

Content-rich dashboard (monitoring)
- CloudWatch (便宜）
- Grafana (好看，可能要 4GB ram)

Badges
- GitHub CICD
- 線上版本
- Discord

Other
- GitHub Sponsors

**Reply by mikaeloduh** [reply]
[GitHub Sponsors · GitHub](https://github.com/sponsors)

**Reply by mikaeloduh** [reply]
範例
![GitHub Sponsors](https://img.shields.io/github/sponsors/Homebrew)

**Comment by mikaeloduh** [comment]
**目前可以放的 Badge：**

[![](https://github.com/Game-as-a-Service/Lobby-Platform-Service/actions/workflows/build-and-deploy.yml/badge.svg)]()

[![Discord](https://img.shields.io/discord/937992003415838761?label=Discord)]() ← 要管理員開才有

[![Swagger Validator](https://img.shields.io/swagger/valid/3.0?specUrl=https%3A%2F%2Fapi.gaas.waterballsa.tw%2Fswagger-ui%2Fapi-docs)](https://api.gaas.waterballsa.tw/swagger-ui/index.html) ← Swagger schema 有錯

[![Website](https://img.shields.io/website?url=https%3A%2F%2Fapi.gaas.waterballsa.tw%2Fhealth&label=service%20status)](https://lobby.gaas.waterballsa.tw)

[![Code of Conduct](https://img.shields.io/badge/code%20of-conduct-ff69b4.svg?style=flat)]()  

[![GitHub Release](https://img.shields.io/github/release/Game-as-a-Service/Lobby-Platform-Service.svg)](https://github.com/Game-as-a-Service/Lobby-Platform-Service/releases)

**Reply by mikaeloduh** [reply]
如果要放 Discord 的話需要管理員開設定 @Johnny850807 
這邊有設定方式 https://shields.io/badges/discord

**Comment by iankingh** [comment]
## Badge ：

### markdown  版

[![](https://github.com/Game-as-a-Service/Lobby-Platform-Service/actions/workflows/build-and-deploy.yml/badge.svg)]()

[![Discord](https://img.shields.io/discord/937992003415838761?label=Discord)]() 

[![Swagger Validator](https://img.shields.io/swagger/valid/3.0?specUrl=https%3A%2F%2Fapi.gaas.waterballsa.tw%2Fswagger-ui%2Fapi-docs)](https://api.gaas.waterballsa.tw/swagger-ui/index.html)

[![Website](https://img.shields.io/website?url=https%3A%2F%2Fapi.gaas.waterballsa.tw%2Fhealth&label=service%20status)](https://lobby.gaas.waterballsa.tw)

[![Code of Conduct](https://img.shields.io/badge/code%20of-conduct-ff69b4.svg?style=flat)]()  

[![GitHub Release](https://img.shields.io/github/release/Game-as-a-Service/Lobby-Platform-Service.svg)](https://github.com/Game-as-a-Service/Lobby-Platform-Service/releases)
### 使用HTML置中版

例如：https://github.com/spaceship-prompt/spaceship-prompt/blob/master/README.md

<p style="text-align:center;">
    <a href="">
        <img src="https://github.com/Game-as-a-Service/Lobby-Platform-Service/actions/workflows/build-and-deploy.yml/badge.svg" alt="badge">
    </a>
    <a href="https://maven-badges.herokuapp.com/maven-central/com.ejlchina/bean-searcher/">
        <img src="https://maven-badges.herokuapp.com/maven-central/com.ejlchina/bean-searcher/badge.svg" alt="Maven Central">
    </a>
    <a href="https://lobby.gaas.waterballsa.tw">
        <img src="https://img.shields.io/website?url=https%3A%2F%2Fapi.gaas.waterballsa.tw%2Fhealth&label=service%20status" alt="Website">
    </a>
    <a href="https://github.com/Game-as-a-Service/Lobby-Platform-Service/releases">
        <img src="https://img.shields.io/github/release/Game-as-a-Service/Lobby-Platform-Service.svg" alt="GitHub Release">
    </a>
    <a href="https://github.com/Game-as-a-Service/Lobby-Platform-Service/blob/main/LICENSE">
        <img src="https://img.shields.io/hexpm/l/plug.svg" alt="License">
    </a>
    <a href="https://github.com/Game-as-a-Service/Lobby-Platform-Service/blob/main/CODE_OF_CONDUCT.md">
        <img src="https://img.shields.io/badge/code%20of-conduct-ff69b4.svg?style=flat" alt="Code of Conduct ">
    </a>
    <a href="https://api.gaas.waterballsa.tw/swagger-ui/index.html">
        <img src="https://img.shields.io/swagger/valid/3.0?specUrl=https%3A%2F%2Fapi.gaas.waterballsa.tw%2Fswagger-ui%2Fapi-docs" alt="Swagger Validator">
    </a>
    <a href="">
        <img src="https://img.shields.io/discord/937992003415838761?label=Discord" alt="Discord">
    </a>
</p>

**Comment by mikaeloduh** [comment]
### 2023/10/24 會議討論：

* 中翻英語意不通順
* 建議未來新增內容以英文版 readme 為主，再翻譯成其他語言


預計本週完成 review PR https://github.com/Game-as-a-Service/Lobby-Platform-Service/pull/160

**Comment by ricksu978** [comment]
Fluent Assertion
https://github.com/fluentassertions/fluentassertions

![image](https://github.com/Game-as-a-Service/Lobby-Platform-Service/assets/35659961/bc2bd552-3300-4cf3-94aa-55ec87e29cd2)
